[AND-2913] MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK

anton-blashchuk opened the pull request.

Commit:
Release Vungle adapters - 6.3.24.4

(cherry picked from commit a38d3cddf5f2253b4230add452d28d2ac0e67e1e)

Commit:
Release Vungle adapters - 6.4.11.3

(cherry picked from commit ccf7f9d08789dfdb59ee418fc6618e0faf8f08d2)

Commit:
Release Vungle adapters - 6.4.11.4

(cherry picked from commit 6f29536adb2955389dbe63f74dd0a67b095551ea)

Commit:
Release Vungle adapters 6.4.11.4

(cherry picked from commit 98edc2babafcd1b19bde3c5e32aef3a34a4ec05d)

Commit:
Release Vungle adapter 6.4.11.4

(cherry picked from commit 9cbb496c508eb40a41438d3099b6c7f149787002)

Commit:
Release Vungle adapters - 6.4.11.5

(cherry picked from commit 566d53ea3ae1b09c5d5e9fe483b19a5bcfdf70f3)

palrahul committed:
Release Vungle adapter 6.5.2.0  (#223)

(cherry picked from commit 821e6c6d8082d4d243b2920786317e1461347f3a)

HuFangshuai committed:
Release Vungle adapter 6.5.3.0 (#243)

(cherry picked from commit 788004185708e7b859f66dc62fc155c09ef33264)

Commit:
[AND-2913] MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK
work in progress

Commit:
Merge branch 'develop' into feature-AND-2913-deAndroidX

# Conflicts:
#	Vungle/CHANGELOG.md
#	Vungle/build.gradle
#	Vungle/src/main/java/com/mopub/mobileads/VungleAdapterConfiguration.java
#	Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java
#	Vungle/src/main/java/com/mopub/mobileads/VungleInterstitial.java
#	Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java
#	Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java
#	Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java
#	Vungle/src/main/java/com/mopub/mobileads/VungleRouterListener.java
#	build.gradle
#	mopub-android-sdk

Commit:
[AND-2913] MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK
hack-hack-hack but make a Banner work

Commit:
[AND-2913] more work in progress, some androidX removed, dependencies reworked, butterknife removed
MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK

Commit:
[AND-2913] slightly reverted VungleBanner to make it work with older reflection; changed UI elements to ones not using AndroidX
MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK

HuFangshuai commented:
@anton-blashchuk-zoolatech please don't merge this PR to master branch.

palrahul commented on Vungle/src/main/java/com/mopub/mobileads/AdData.kt:
```diff
@@ -0,0 +1,194 @@
+import kotlin.collections.HashMap
+
+@Parcelize
+data class AdData(
```
@anton-blashchuk-zoolatech where are these Kotlin classes from, are they ported from some version of MoPub's SDK code

HuFangshuai replied:
I think this class not related to non-AndroidX patch issue. Can we remove it and Kotlin related gradle configration?

anton-blashchuk replied:
done

palrahul commented:
@anton-blashchuk-zoolatech suggestion to checkout target branch say 6.7.0-nax-release from 6.7.0-GA tag , and then create this PR from your dev branch into it

Commit:
[AND-2913] banner with header bidding fixed
MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK

Commit:
[AND-2933] updated calculation for AdSize knowing that mopub-sdk will not pass proper adunit_format
MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK

Commit:
[AND-2933] removed Kotlin from project
MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK

anton-blashchuk commented:
> @anton-blashchuk-zoolatech please don't merge this PR to master branch.

yes, sure, that was unintended - I think I forgot to update target branch at that point.

HuFangshuai commented on Vungle/src/main/java/com/mopub/mobileads/VungleAdapterConfiguration.java:
```diff
@@ -42,8 +42,7 @@ public String getAdapterVersion() {
     public String getBiddingToken(@NonNull Context context) {
-        final String bidToken = Vungle.getAvailableBidTokens(10);
-        return bidToken;
+        return null;
```
Why removed this bid token logic?

Commit:
[AND-2933] reverted deletion of bid token logic
MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK

alexander-gubov commented on Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:
```diff
@@ -257,60 +266,50 @@ private boolean validateIdsInServerExtras(Map<String, String> serverExtras) {
-                        if (mInteractionListener != null) {
-                            mInteractionListener.onAdClicked();
+                        if (mCustomEventBannerListener != null) {
+                            mCustomEventBannerListener.onBannerClicked();
```
we can't call mCustomEventBannerListener.onBannerClicked() in any case, there were params in onAdEnd() to filter but logic was moved to onAdClick()

ct-anton committed:
[AND-2933] updated subproject hashes

palrahul commented on Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java:
```diff
@@ -237,14 +244,24 @@ private void modifyAdConfig(AdConfig adConfig, VungleMediationConfiguration medi
+                // Vungle does not provide a callback when a user should be rewarded.
+                // You will need to provide your own reward logic if you receive a reward with
+                // "NO_REWARD_LABEL" && "NO_REWARD_AMOUNT"
+                MoPubRewardedVideoManager.onRewardedVideoCompleted(VungleRewardedVideo.class,
```
shouldnt this be in onAdRewarded

palrahul commented on Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java:
```diff
@@ -237,14 +244,24 @@ private void modifyAdConfig(AdConfig adConfig, VungleMediationConfiguration medi
+                // Vungle does not provide a callback when a user should be rewarded.
+                // You will need to provide your own reward logic if you receive a reward with
+                // "NO_REWARD_LABEL" && "NO_REWARD_AMOUNT"
+                MoPubRewardedVideoManager.onRewardedVideoCompleted(VungleRewardedVideo.class,
```
as there is separate onRewardedVideoClosed. , should the onRewardedVideoCompleted be called from onAdRewarded.

HuFangshuai replied:
We can remove code parts from line 253 to line 261.

palrahul commented on Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java:
```diff
@@ -253,27 +270,17 @@ public void onAdEnd(String placementId) {
-                            MoPubReward.NO_REWARD_AMOUNT));
-                }
-            }
+            MoPubRewardedVideoManager.onRewardedVideoClicked(VungleRewardedVideo.class,
```
not sure if this belongs here

HuFangshuai replied:
Pls replace this event implementaion with following code, we should use `onRewardedVideoCompleted() `:
```
if (mPlacementId.equals(placementId)) {
	MoPubLog.log(getAdNetworkId(), SHOULD_REWARD, ADAPTER_NAME, MoPubReward.NO_REWARD_AMOUNT,
	        MoPubReward.NO_REWARD_LABEL);
	// Vungle does not provide a callback when a user should be rewarded.
	// You will need to provide your own reward logic if you receive a reward with
	// "NO_REWARD_LABEL" && "NO_REWARD_AMOUNT"
	MoPubRewardedVideoManager.onRewardedVideoCompleted(VungleRewardedVideo.class,
	        mPlacementId,
	        MoPubReward.success(MoPubReward.NO_REWARD_LABEL,
	                MoPubReward.NO_REWARD_AMOUNT));
}
```

HuFangshuai commented on Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:
```diff
@@ -23,21 +21,22 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
-import static com.mopub.common.DataKeys.ADUNIT_FORMAT;
+import static com.mopub.network.ImpressionData.ADUNIT_FORMAT;
```
Pls use `import static com.mopub.common.DataKeys.ADUNIT_FORMAT;`

anton-blashchuk replied:
it's not there to use.

HuFangshuai replied:
yes, MoPub 5.8.0 SDK not contains `DataKeys.ADUNIT_FORMAT`.

HuFangshuai commented on Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:
```diff
@@ -121,8 +130,8 @@ public void run() {
-        VungleMediationConfiguration.adConfigWithExtras(mAdConfig, extras);
-        if (VungleMediationConfiguration.isStartMutedNotConfigured(extras)) {
+        VungleMediationConfiguration.adConfigWithLocalExtras(mAdConfig, localExtras);
+        if (VungleMediationConfiguration.isStartMutedNotConfiguredLocal(localExtras)) {
```
There's no difference between `isStartMutedNotConfiguredLocal (localExtras)` and `isStartMutedNotConfigured(localExtras)`. Suggestion to reuse it.
```
if (VungleMediationConfiguration.isStartMutedNotConfigured(localExtras)) {
```

anton-blashchuk replied:
they have different parameter type. I'll change type for isStartMutedNotConfigured's parameter to Map<String, Object>, and remove isStartMutedNotConfiguredLocal

HuFangshuai commented on Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:
```diff
@@ -148,30 +157,36 @@ public void run() {
-            if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
-                adSizeType = VUNGLE_MREC;
-            }
+        if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
```
Pls keep `isMRECFormat` logic here, `adWidth` and `adHeight` match VUNGLE_MREC not means this ad unit is MREC ad format.
```
final boolean isMRECFormat = "medium_rectangle".equals(adUnitFormat);
        if (isMRECFormat) {
            if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
                adSizeType = VUNGLE_MREC;
            }
        } else {
        ...
        }
```

alexander-gubov replied:
@HuFangshuai The issue is that "medium_rectangle" is not passed in old Mopub as far as I remember.

HuFangshuai replied:
Thank @alexander-gubov, yes, MoPub 5.8.0 SDK does not pass `medium_rectangle ` to adapter.

HuFangshuai commented on Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java:
```diff
@@ -120,11 +120,53 @@ static void adConfigWithExtras(@NonNull final AdConfig adConfig,
         }
     }
 
+    static void adConfigWithLocalExtras(AdConfig adConfig, Map<String, Object> localExtras) {
```
Seems this method same as `adConfigWithLocalExtras()`, can we reuse it?

anton-blashchuk replied:
almost same except parameter. I'll just remove adConfigWithExtras which needs values in Map to be Strings

HuFangshuai commented on Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java:
```diff
@@ -120,11 +120,53 @@ static void adConfigWithExtras(@NonNull final AdConfig adConfig,
                 !extras.containsKey(Builder.EXTRA_SOUND_ENABLED_KEY);
     }
 
+    static boolean isStartMutedNotConfiguredLocal(Map<String, Object> localExtras) {
```
Same as `isStartMutedNotConfigured ()`, suggestion to reuse it.

anton-blashchuk replied:
again different non-exchangeable(one way) parameter type.

Commit:
[AND-2933] replaced mopub-sdk dependency - was module from sources now built-before aar; addressed some of peer review comments
MoPub: Wrap-up Build non-AndroidX version of adapter with 6.7.0 SDK
https://vungle.atlassian.net/browse/AND-2933

Commit:
[AND-2933] addressed comments in peer review

Commit:
[AND-2933] removed test dependencies, updated mediation/Vungle version

alexander-gubov committed:
[AND-2913] remove including  mopubsdk

HuFangshuai reviewed (approved):
LGTM

palrahul reviewed (approved):
lgtm
